### PR TITLE
produce error when CHPL_LOCALE_MODEL=gpu and CHPL_COMM != none

### PR DESCRIPTION
**Pull request by stonea**
This is for: https://github.com/chapel-lang/chapel/issues/18858

As shown here: #18850 GPU code generation does not work if CHPL_COMM is something other than none.

As of the release if you do this you'll get a confusing error message about a missing fatbin file at runtime. Ultimately we'll want to support this combination but it seems worth producing a nice error message in the meantime.

This PR also modifies things so if the "GPU code generation process" that we fork off from the main process encounters an error we stop the main process once we join back (otherwise we'll always get an additional error about a missing `.fatbin` file when the GPU thread encounters an issue)

### CODE REVIEW

File: test/gpu/native/environment/gasnet.compopts [hunk 1]
```diff
@@ -0,0 +1 @@
+--comm gasnet
```
Comment (e-kayrakli) [inline test/gpu/native/environment/gasnet.compopts:1]: Just FYI, we have `.compenv` files. So, what you actually need here is `gasnet.compenv` file that has `CHPL_COMM=gasnet` in it.

Or that and this.. But I'm fine as-is

File: util/chplenv/printchplenv.py [hunk 2]
```diff
@@ -211,6 +212,11 @@ def compute_all_values():
     chpl_llvm.validate_llvm_config()
     chpl_compiler.validate_compiler_settings()
 
+    if ENV_VALS['CHPL_LOCALE_MODEL'] == 'gpu' and \
+       ENV_VALS['CHPL_COMM'] != "none":
+        error("The prototype GPU support does not work when CHPL_COMM is not set to\n\"none\".");
```
Comment (e-kayrakli) [inline util/chplenv/printchplenv.py:217]: Are there any issues with having a `chpl_gpu.validate(comm)` function? I feel like this is unnecessarily inconsistent with similar checks.

Comment (stonea) [reply util/chplenv/printchplenv.py:217]: The reason I didn't do that is I wasn't sure if this really belongs under `chpl_locale_model` or `chpl_comm` (the issues really about an interaction between the two). I hadn't thought about `chpl_gpu` but it isn't any more obvious that it belongs there.

But I agree having the code floating there is kind of ugly. I think I'll take your suggestion and stick it in `chpl_gpu` on the basis that even if that doesn't seem like the "perfect" place for it, it still seems less ugly than having it inlined there.

Comment (e-kayrakli) [reply util/chplenv/printchplenv.py:217]: Yeah good point about the place in which that validation belongs. But, yeah, I agree that it is less ugly.

File: test/gpu/native/environment/gasnet.compopts (hunk 1 above)
Comment (stonea) [reply test/gpu/native/environment/gasnet.compopts:1]: If we really wanted to be exhaustive about testing we should probably do both. I think I'll leave it as is. But thanks for letting me know about `.compenv` (I wasn't aware of it).

**Pull request merged**